arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -163,6 +163,69 @@ def rule_fr_nl_1_01(
                         return
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18
+    ],
+)
+def rule_fr_nl_2_04(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-2.04: The 'link:schemaRef' element MUST NOT appear more than once
+    """
+    schema_ref_model_objects = []
+    for doc in val.modelXbrl.urlDocs.values():
+        if doc.type == ModelDocument.Type.INSTANCE:
+            for refDoc, docRef in doc.referencesDocument.items():
+                if docRef.referringModelObject.localName == "schemaRef":
+                    schema_ref_model_objects.append(docRef.referringModelObject)
+    if len(schema_ref_model_objects) > 1:
+        yield Validation.error(
+            codes='NL.FR-NL-2.04',
+            msg=_('The \'link:schemaRef\' element must not appear more than once.'),
+            modelObject=schema_ref_model_objects
+        )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18
+    ],
+)
+def rule_fr_nl_2_05(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-2.05: The 'link:linkbaseRef' element MUST NOT occur
+    """
+    linkbase_ref_model_objects = []
+    for doc in val.modelXbrl.urlDocs.values():
+        if doc.type == ModelDocument.Type.INSTANCE:
+            for refDoc, docRef in doc.referencesDocument.items():
+                if docRef.referringModelObject.localName == "linkbaseRef":
+                    linkbase_ref_model_objects.append(docRef.referringModelObject)
```

Comment:
Same as other comment. Maybe yield error on each discovery and provide file name and line number if available?

Reply:
Check my note above. I feel like this is another example where all linkbase refs should be highlighted at once so that they can be removed at once.